### ISSUE TITLE
Improve exception safety with smart pointers

### ISSUE THREAD

**Issue by elfring** [Harha/MirageRender#3]
Would you like to wrap any pointer data members with the template class "[std::unique_ptr](https://en.wikipedia.org/wiki/Smart_pointer#unique_ptr)"?

Update candidates:
- [Material](https://github.com/Harha/MirageRender/blob/55024ea404d664a0e2720d536b898ab831496793/src/core/material.cpp#L27)
- [Scene](https://github.com/Harha/MirageRender/blob/55024ea404d664a0e2720d536b898ab831496793/src/core/scene.cpp#L21)

**Comment by Harha** [comment]
Yes, things would make way more sense that way. Though I'd have to read more about smart pointers before doing anything like this. I'm a beginner with C++, I know the basics but I haven't had the time to really read about the language and it's standard library that much, most of what I know I've just learned by experimentation.

I will do it, or if you want to I'm always happily accepting pull requests.

Anyways, thanks a lot. These are some nice suggestions you are sending.
